tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_argmax_padding.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import pytest
+import tt_lib
+from loguru import logger
+from tests.tt_eager.python_api_testing.sweep_tests import comparison_funcs
+
+
+@pytest.mark.parametrize(
+    "input_shapes",
+    (
+        (torch.Size([1, 1, 1, 10])),
+        (torch.Size([1, 1, 10, 20])),
+        (torch.Size([1, 1, 30, 3])),
+        (torch.Size([1, 4, 3, 5])),
+        (torch.Size([5, 4, 3, 20])),
+    ),
+)
+@pytest.mark.parametrize("dim", (3, 2, 1, 0, -1, -2, -3, -4))
+@pytest.mark.parametrize("all", (True, False))
+class TestArgmax:
+    def test_argmax(self, input_shapes, dim, all, device):
+        torch.manual_seed(10)
+        input_data = torch.randn(input_shapes).bfloat16()
+        input_tensor = (
+            tt_lib.tensor.Tensor(input_data, tt_lib.tensor.DataType.BFLOAT16)
+            .pad_to_tile(100)
+            .to(tt_lib.tensor.Layout.TILE)
+            .to(device)
+        )
+        tt_output_tensor_on_device = tt_lib.tensor.argmax(input_tensor, dim=dim, all=all)
+        tt_out_tensor = tt_output_tensor_on_device.cpu().to(tt_lib.tensor.Layout.ROW_MAJOR).to_torch()
+        if all:
+            golden_tensor = torch.argmax(input_data)
+            tt_out_tensor = tt_out_tensor[0, 0, 0, 0]
+        else:
+            golden_tensor = torch.argmax(input_data, dim=dim)
+            if dim == 1 or dim == -3 or dim == 0 or dim == -4:
+                tt_out_tensor = tt_out_tensor[0, :, 0 : input_shapes[2], 0 : input_shapes[3]]
+            else:
+                if input_shapes[1] != 1 or input_shapes[0] != 1:
+                    if dim == 2 or dim == -2:
+                        tt_out_tensor = tt_out_tensor[0, :, :, 0 : input_shapes[3]]
+                    else:
+                        tt_out_tensor = tt_out_tensor[0, :, :, 0 : input_shapes[2]]
+                else:
+                    if dim == 2 or dim == -2:
+                        tt_out_tensor = tt_out_tensor[0, 0, 0, 0 : input_shapes[3]]
+                    else:
+                        tt_out_tensor = tt_out_tensor[0, 0, 0, 0 : input_shapes[2]]
+        pt_out_tensor = golden_tensor
+        comp_pass, comp_out = comparison_funcs.comp_pcc(pt_out_tensor, tt_out_tensor, pcc=0.99)
+        comp_all, _ = comparison_funcs.comp_allclose(pt_out_tensor, tt_out_tensor, atol=4, rtol=1e-1)
```

Comment:
```suggestion
        comp_all, _ = comparison_funcs.comp_allclose(pt_out_tensor, tt_out_tensor, atol=0, rtol=0)
```
Since `argmax` is an operation that returns an index, both `atol` and `rtol` need to be set to 0. 
An error in the index is a much bigger issue than an error in floats. 
For example, in a classification problem where the decision is between "road" and "car", an error in the index could lead to a wrong judgment between a car and a road, potentially causing a car accident.

The test case where the problem occurs is as follows:
- input_shape: torch.Size([5, 4, 3, 20])
- dim: 3
- all: True

In this case, the correct value returned by torch is 1025, but tt is returning 1024. This seems to be an issue because the output dtype is `bfloat16`.

I am not sure if this will be fixed in this PR, but to prevent such problems, we need to quickly modify the `argmax` output type to integer.
@tt-aho @razorback3 @umadevimcw

Reply:
@hschoi4448 YThe index mismatch is because of the bfloat16 as discussed here https://github.com/tenstorrent/tt-metal/issues/4787. I will try to fix this